### ISSUE TITLE
Base aspnet image in .NET 8 has an UTC issue

### ISSUE THREAD

**Issue by ydubernet** [dotnet/dotnet-docker#5274]
### Describe the Bug
As I'm migrating some .NET apps to docker, I came across a time issue.

Within my app, we connect to an authentication server. The authentication token gets generated and is valid starting a given UTC date which is the current UTC date of the authentication request.

I then execute an HTTP request with the generated bearer token but the authentication server answers to my docker app  : `Bearer error="invalid_token", error_description="The token is not valid before '03/15/2024 16:35:40'"`.

Therefore I ended up investigating the UTC hour of my container and it appears that the time displayed within my .NET container is said to be UTC but is actually UTC -5:00. To be sure it wasn't coming from my code, I reproduced it with a HelloWorld.NET app.

### Steps to Reproduce

To reproduce, I created a HelloWorld app in .NET 8 and created a minimal Dockerfile to build a docker image executing my app.

> FROM mcr.microsoft.com/dotnet/aspnet:8.0 AS base
> USER app
> WORKDIR /app
> 
> FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
> ARG BUILD_CONFIGURATION=Release
> 
> WORKDIR /src
> COPY [ "HelloWorld.NET.csproj", "."]
> RUN dotnet restore "./HelloWorld.NET.csproj"
> COPY . .
> RUN dotnet build "./HelloWorld.NET.csproj" -c $BUILD_CONFIGURATION -o /app/build
> 
> FROM build AS publish
> ARG BUILD_CONFIGURATION=Release
> RUN dotnet publish "./HelloWorld.NET.csproj" -c $BUILD_CONFIGURATION -o /app/publish /p:UseAppHost=false
> 
> FROM base AS final
> WORKDIR /app
> COPY --from=publish /app/publish .
> ENTRYPOINT ["dotnet", "HelloWorld.NET.dll"]

I would then run `docker run helloworldnet:dev` and log into my container with a `docker exec -ti _yourcontainerid_ /bin/bash`.
Then I would run the command `date` and here is the result :

![image](https://github.com/dotnet/dotnet-docker/assets/10231598/ae05a7f2-aae3-4421-b067-30e654110f16)

I checked on github regarding maybe already existing issues and found https://github.com/dotnet/runtime/issues/62545 which seems to talk about a similar issue since .NET 6.
Reading the thread, I then ran `cat /etc/timezone` and the result is `Etc/UTC`.

Am I missing something or from the base image of mcr.microsoft.com/dotnet/aspnet:8.0, we need to do some manual edits in our Dockerfile which would let the server consider UTC as the actual UTC and not UTC-5:00 ?

Thank you a lot for the time you'll give.

_FYI my timezone is currently CET, so UTC+1:00._

### Output of `docker version`

Client:
 Cloud integration: v1.0.22
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Mon Dec 13 11:44:07 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.5.1 (74721)
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Mon Dec 13 11:43:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

### Output of `docker info`

Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.7.1)
  compose: Docker Compose (Docker Inc., v2.2.3)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 3
  Running: 2
  Paused: 0
  Stopped: 1
 Images: 18
 Server Version: 20.10.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.72-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 24.78GiB
 Name: docker-desktop
 ID: RJUJ:AFIC:3IOP:7K7X:I5XU:GIQL:TFMP:3AJV:RFPN:QCYR:XPTU:Y5WJ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: **someproxy**
 HTTPS Proxy: **someproxy**
 No Proxy: hubproxy.docker.internal,localhost,**some internal root domains**,127.0.0.1
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support

**Comment by richlander** [comment]
What I see on my machine.

```bash
rich@mazama:~$ cat /etc/os-release | head -n 1
PRETTY_NAME="Ubuntu 23.10"
rich@mazama:~$ cat /etc/timezone
America/Los_Angeles
rich@mazama:~$ docker run --rm mcr.microsoft.com/dotnet/aspnet:8.0 cat /etc/timezone
Etc/UTC
rich@mazama:~$ docker run --rm debian:bookworm-slim cat /etc/timezone
Etc/UTC
```

**Comment by omajid** [comment]
As I understand it, the container uses the system clock, assumes it is in UTC, then uses timezone defined in the container to convert it to the local time. 

You can set the environment variable `TZ` to tell your container what timezone to use:

```Dockerfile
ENV TZ=America/Toronto
```

Or pass it as a command line argument to `docker run`.  Or even [share the timezone from the OS as a volume mount](https://stackoverflow.com/questions/57607381/how-do-i-change-timezone-in-a-docker-container).

On Linux + docker (well, podman), I can confirm the time is actually correct:

```
# show the local time in my host OS
$ LANG=C date
Fri Mar 15 14:37:17 EDT 2024

# show the local time in a container
$ docker run -e LANG=C mcr.microsoft.com/dotnet/aspnet  date
Fri Mar 15 18:38:49 UTC 2024


# show the local time converted to UTC on my host OS
$ LANG=C TZ=UTC date
Fri Mar 15 18:30:26 UTC 2024

# show the local time converted to UTC in the docker container
$ docker run -e LANG=C -e TZ=UTC mcr.microsoft.com/dotnet/aspnet  date
Fri Mar 15 18:30:57 UTC 2024
```

In other words, the host and container both agree on the actual time, just not the timezone, and so display it differently.

In your authentication message/error, is a timezone part of the request/response? Does the app just assume that the time is always in a certain timezone, or does it include timezone in data/calculations/authenticatio?

I know Windows and Linux make different assumptions about the system clock (the hardware one). Windows assumes it's in the local timezone, while Linux assumes it's in UTC. That couldn't be an issue on WSL2, right?

**Comment by richlander** [comment]
We also have a doc dedicated to this general topic: https://github.com/dotnet/dotnet-docker/tree/main/samples/globalapp

**Comment by ydubernet** [comment]
Hello,

Thank you for your help on that issue.

@omajid for the authentication message/error, it didn't show me the timezone as part of the request/response.
And I did think about the local timezone vs UTC issue but then I couldn't understand why the jetlag would be 6 hours, not just 1 hour as my timzone is UTC +1:00.

However, as I wanted to reproduce it to be double check the error message, I think it may have been a temporary issue as, this morning, I'm not able to reproduce it.
![image](https://github.com/dotnet/dotnet-docker/assets/10231598/99f7db25-f034-47cd-97bf-f211e03e6681)

I'll definitely keep this issue and all your information in mind is case I reproduce it.
